core/components/molecules/modalHeader/ModalHeader.tsx

```diff
@@ -7,30 +7,50 @@ export interface ModalHeaderProps extends BaseProps {
   heading: string;
   onClose: (event: React.MouseEvent<HTMLElement, MouseEvent>, reason?: string) => void;
   subHeading?: string;
+  seperator?: boolean;
+  backIcon?: boolean;
+  backIconCallback?: (e: React.MouseEvent<HTMLElement, MouseEvent>) => void;
 }
 
 export const ModalHeader = (props: ModalHeaderProps) => {
-  const { className, heading, subHeading, onClose } = props;
+  const { className, heading, subHeading, onClose, seperator, backIcon, backIconCallback } = props;
   const baseProps = extractBaseProps(props);
 
   const classes = classNames({
-    'Modal-header': true
+    'Modal-header': true,
+    ['Modal-header--backIcon']: backIcon,
+    ['Modal-header--seperator']: seperator
   }, className);
 
+  const wrapperClass = classNames({
+    'Modal-headerWrapper': true,
+    ['Modal-headerWrapper--backIcon']: backIcon,
+  });
+
   return (
     <div data-test="DesignSystem-ModalHeader" {...baseProps} className={classes}>
-      <div className="Modal-headerHeading">
-        <Heading>{heading}</Heading>
-        <Icon
-          size={20}
-          name={'close'}
-          className={'mx-2 cursor-pointer'}
-          data-test="DesignSystem-ModalHeader--CloseIcon"
-          onClick={(event: React.MouseEvent<HTMLElement, MouseEvent>) => onClose(event, 'IconClick')}
-        />
+      <div className={wrapperClass}>
+        {backIcon && (
+          <div
+            className="Modal-headerBackIcon mr-4 cursor-pointer"
+            onClick={backIconCallback}
+          >
+            <Icon name="keyboard_backspace" size={20} />
+          </div>
+        )}
+        <div className="Modal-headerHeading">
+          <Heading>{heading}</Heading>
+          <Icon
+            size={20}
+            name={'close'}
+            className={'mx-2 cursor-pointer'}
+            data-test="DesignSystem-ModalHeader--CloseIcon"
+            onClick={(event: React.MouseEvent<HTMLElement, MouseEvent>) => onClose(event, 'IconClick')}
+          />
+        </div>
       </div>
       {subHeading && (
-        <Text data-test="DesignSystem-ModalHeader--Subheading" appearance="subtle" className="mt-2">
+        <Text data-test="DesignSystem-ModalHeader--Subheading" appearance="subtle" className="mt-2 ml-7">
```

Comment:
`ml-7` is being applied even when `backIcon: false`. Apply conditional class.

Reply:
As per design, `Subheading` will always have a `24px` (ml-7) margin-left.